### DISCUSSION TITLE
Modifying Existing IFC to add chainage info

### DISCUSSION THREAD

**Discussion by CJAndrew84** [GeometryGym/GeometryGymIFC#68] [Q&A]
Does any one have any examples of a way to determine the chainage and offset of an IFCElement relative to an IfcAlignment?
Basically, the reverse of placing a new object by distance along an alignment.

**Comment by jmirtsch** [comment]
Hi Chris,
At present this is out of scope of the Geometry Gym IFC toolkit (which is primarily serialization of the IFC data model).  There are a few barriers and logistic challenges.
Geometry Gym has closed source developments that extend the open source code, but we do not have our own geometry kernel.  Most developments are in the form of plugins, and utilize the geometry API provided in the host application.  
I haven't found any .net classes that facilitate basic geometry (like points and vectors) with double precision (which is needed for infrastructure projects in grid coordinates).  We typically use Rhinocommon ( https://developer.rhino3d.com/guides/rhinocommon/ ) but to pursue this type of functionality,  it would be nice if the code used dotnet core.
The final logistic is resources.  To facilitate this improvement, external contributions or funding would be required.  To implement properly, acknowledging station equations would be required.  
I don't know of alternative Open Source IFC projects that provides this (certainly not .net).  At present the best I could suggest is importing into an application supporting IFC and injecting it into the IFC.  

Look forward to discussing further.

Jon

**Reply by CJAndrew84** [reply]
Worth asking. We have a Excel VBA script that can get chainage offset for any point, it uses x,y points at 1m centers along the alignment.

I might try to do something similar in c#
